build/build.js

```diff
@@ -0,0 +1,112 @@
+const rollup = require('rollup');
+const typescript = require('rollup-plugin-typescript2');
+const rimraf = require('rimraf');
+
+const LIB_NAME = 'aurelia-store';
+const cacheRoot = '.rollupcache';
+const externalLibs = [
+  'aurelia-dependency-injection',
+  'aurelia-event-aggregator',
+  'aurelia-logging',
+  'aurelia-pal',
+  'aurelia-framework',
+  'rxjs',
+  'rxjs/operators'
+];
+
+clean().then(build).then(generateDts);
+
+/**
+ * @type {() => Promise<Error | null>}
+ */
+function clean() {
+  console.log('\n==============\nCleaning dist folder...\n==============');
+  return new Promise(resolve => {
+    rimraf('dist', (error) => {
+      if (error) {
+        throw error;
+      }
+      resolve();
+    });
+  });
+}
+
+function generateDts() {
+  console.log('\n==============\nGenerating dts bundle...\n==============');
+  return new Promise(resolve => {
+    const ChildProcess = require('child_process');
+    ChildProcess.exec('npm run bundle-dts', (err, stdout, stderr) => {
+      if (err || stderr) {
+        console.log('Generating dts error:');
+        console.log(stderr);
+      } else {
+        console.log('Generated dts bundle successfully');
+        console.log(stdout);
+      }
+      resolve();
+    });
+  });
+};
+
+function build() {
```

Comment:
can we split this one up to several smaller functions and having build consume them all, that way independent builds would be possible. The reason is that on a slower machine you're definitely not a fan of having to build all the versions during development if only one specific is needed.

Reply:
I can try this, it sounds doable